File: 프로그래머스/lv2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations)
{
    sort(citations.rbegin(), citations.rend());
    int answer = 0;
    for (int i = 0; i < citations.size(); i++)
    {
        if (i + 1 <= citations[i])
            answer++;
    }
    
    return answer;
}   

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {      
        Arrays.sort(citations);
        
        for (int h = citations.length; h >= 0; h--) {
            int cnt = 0;
            
            for (int j = citations.length - 1; j >= 0; j--) {
                if (citations[j] >= h) cnt++;
                else break;
            }
            
            if (h <= cnt) return h;
        
        }
        
        
        return 0;
    }
}